### DISCUSSION TITLE
How can I run the json editor in a read only mode from the start?

### DISCUSSION THREAD

**Discussion by sabinem** [zauberzeug/nicegui#3047] [Q&A] [answered]
### Question

My goal is to render the ui.json_editor in read only mode from the start, but I struggle to achieve this.

Below is my code: 

- the editor is not in ready only mode initially, even though I tried to set it up that way
- when I press the button `Readonly` it will go into readonly mode
- when I compare the `editor.__dict__` I don't see any difference between the initial set up and the set up after the `Readonly` button was pressed. 

Any help on how can I achieve my goal would be highly appreciated. Thanks in advance for your help with this.

```py
from pprint import pprint
from nicegui import ui

editor = ui.json_editor(
    {'content': {'json': current_execution_to_add}},
     {'context':{'readOnly': True}}
)
ui.button('Readonly', on_click=lambda: editor.run_editor_method('updateProps', {'readOnly': True}))
ui.button('print', on_click=lambda: pprint(editor.__dict__))
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @sabinem,

This seems to work:
```py
ui.json_editor({
    'content': {'json': {'A': 1, 'B': 2}},
    'readOnly': True,
})
```

**Reply by sabinem** [reply]
@falkoschindler Thanks a lot. This works indeed.
